packages/js/src/utils/id-token.test.ts

```diff
@@ -0,0 +1,90 @@
+import { SignJWT, generateKeyPair } from 'jose';
+import { StructError } from 'superstruct';
+
+import { decodeIdToken } from './id-token';
+
+describe('decodeIdToken', () => {
+  test('decoding valid JWT should return claims', async () => {
+    const { privateKey } = await generateKeyPair('RS256');
+    const jwt = await new SignJWT({})
+      .setProtectedHeader({ alg: 'RS256' })
+      .setIssuer('foo')
+      .setSubject('bar')
+      .setAudience('qux')
+      .setExpirationTime(2000)
+      .setIssuedAt(1000)
+      .sign(privateKey);
+    const idTokenClaims = decodeIdToken(jwt);
+    expect(idTokenClaims).toEqual({
+      iss: 'foo',
+      sub: 'bar',
+      aud: 'qux',
+      exp: 2000,
+      iat: 1000,
+    });
+  });
```

Comment:
as i mentioned we also need to test if the provider returns more claims than we defined

Reply:
Added a new test case 'decoding valid JWT with non-predefined claims should return all claims', PTAL.